src/frontc/cparser.mly

```diff
@@ -956,7 +956,20 @@ static_assert_declaration:
 ;
 init_declarator_list:                       /* ISO 6.7 */
     init_declarator                              { [$1] }
-|   init_declarator COMMA init_declarator_list   { $1 :: $3 }
+|   init_declarator COMMA init_declarator_attr_list   { $1 :: $3 }
```

Comment:
I guess if Frama-C uses this, it should be fine, but I'm now curious about how the first `init_declarator` is handled differently now: it cannot have attributes while all the others can. I guess you can still write attributes before the first declarator, but aren't those attributes of the type itself, not of the particular variable being declared? And if that's part of the type, then that applies to all the variables being declared, no?

Reply:
> An attribute specifier list may appear immediately before a declarator (other than the first) in a comma-separated list of  declarators in a declaration of more than one identifier using a single list of specifiers and qualifiers. Such attribute specifiers apply only to the identifier before whose declarator they appear.

https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html

So GCC also treats the first element differently here. If one wants to attach an attribute to the first declaration, one can put the entire thing into parens, and then it is unambiguous once again.

And yes, I think any attributes appearing there are then considered to be part of the type. But tbh, after reading this section of the GCC manual, it seems to me like this entire attribute business is a huge mess. Luckily, we don't really do much with most of them.

Reply:
Maybe just add a TODO there with some link, so if attributes on the first declaration ever become an issue, we know we don't properly handle them.